tests/functional/adapter/concurrent_transaction/test_concurrent_transaction.py

```diff
@@ -0,0 +1,128 @@
+import pytest
+import threading
+
+from dbt.tests.util import run_dbt
+from dbt.tests.adapter.simple_seed.test_seed import SeedConfigBase
+from tests.functional.adapter.concurrent_transaction.fixtures import *
+
+
+class BaseConcurrentTransaction(SeedConfigBase):
+    def __init__(self):
```

Comment:
If `query_state` is always set in `def setUp()`, I don't think this is needed.

Reply:
I thought so, but this test failed because it wasn't initialized: https://github.com/dbt-labs/dbt-redshift/actions/runs/4199862373/jobs/7285257800

Reply:
Is that possibly related to the `autouse=True` piece? I was seeing similar behavior until I turned that on.

Reply:
I'll try that and post back.

Reply:
DIdn't help.  AttributeError: 'TestTableConcurrentTransaction' object has no attribute 'query_state'
in the latest test run failure. I don't see `autouse` used in other tests and it seems to have worked fine.

Reply:
When I try to run this locally:

```
(venv) mikealfare@dbtlabs dbt-redshift % python -m pytest tests/functional/adapter/concurrent_transaction/test_concurrent_transaction.py -v
```

I get this:

```
collected 0 items

=== warnings summary ====
tests/functional/adapter/concurrent_transaction/test_concurrent_transaction.py:84
  /Users/mikealfare/Source/dbt-redshift/tests/functional/adapter/concurrent_transaction/test_concurrent_transaction.py:84: PytestCollectionWarning: cannot collect test class 'TestTableConcurrentTransaction' because it has a __init__ constructor (from: tests/functional/adapter/concurrent_transaction/test_concurrent_transaction.py)
```

If you need that attribute to exist, can you maybe make it a class attribute then? Otherwise I don't think these tests will run, and then they'll look like they pass as a a result.